torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3007,10 +3021,164 @@ struct to_ir {
+      auto expr_list = List<Expr>::create(apply.range(), {ll});
+      // Change `apply` to a new Apply node holding a list of
+      // tuples
+      apply = Apply::create(
+          apply.range(), apply.callee(), expr_list, apply.attributes());
+    }
+
+    // If we have kwargs to include, we'll take a similar approach
+    // to the above logic and standardize the Apply node
+    if (!apply.attributes().empty()) {
+      std::vector<Expr> exprs;
+      // Gather all the existing tuples in the input iterable
+      if (!apply.inputs().empty()) {
+        auto tuple_list = ListLiteral(apply.inputs()[0]).inputs();
+        for (auto tuple : tuple_list) {
+          exprs.push_back(tuple);
+        }
+      }
+      // Create tuples out of each kwarg and gather them as well
+      for (auto attr : apply.attributes()) {
+        auto k = StringLiteral::create(apply.range(), attr.name().name());
+        auto v = attr.value();
+        auto tuple_inputs = List<Expr>::create(apply.range(), {k, v});
+        auto tuple = TupleLiteral::create(apply.range(), tuple_inputs);
+        exprs.push_back(tuple);
+      }
+      auto expr_list = List<Expr>::create(apply.range(), {exprs});
+      auto ll = ListLiteral::create(apply.range(), expr_list);
+      auto new_inputs = List<Expr>::create(apply.range(), {ll});
+      auto new_kwargs = List<Attribute>::create(apply.range(), {});
+      apply = Apply::create(
+          apply.range(), apply.callee(), new_inputs, new_kwargs);
+    }
+
+    checkApplyNumInputs(apply, 1);
+
+    auto iter_input = emitSugaredExpr(apply.inputs()[0], 1);
+    const std::string& iter_name = createTempName("$_iter");
+    const std::string& key_name = createTempName("$_key");
+    const std::string& value_name = createTempName("$_value");
+
+    auto key = Var::create(
+        apply.range(), Ident::create(apply.range(), key_name));
+    auto value = Var::create(
+        apply.range(), Ident::create(apply.range(), value_name));
+    auto target = TupleLiteral::create(apply.range(), List<Expr>::create(apply.range(), {key, value}));
+    auto iter = Var::create(
+        apply.range(), Ident::create(apply.range(), iter_name));
+
+    environment_stack->setSugaredVar(
+        apply.range(),
+        iter_name,
+        iter_input,
+        /*annotated_type=*/nullptr);
+
+    auto dc = DictComp::create(apply.range(), key, value, target, iter);
+    auto dc_node = emitDictComprehension(dc, type_hint);
+    NamedValue self = NamedValue(dc.range(), "self", dc_node);
+
+    for (auto kwarg : apply.attributes()) {
+      auto name = StringLiteral::create(kwarg.range(), kwarg.name().name());
+      auto k = emitExpr(name);
+      auto v = emitExpr(kwarg.value());
+      NamedValue input_k = NamedValue(kwarg.range(), "", k);
+      NamedValue input_v = NamedValue(kwarg.range(), "", v);
+    emitBuiltinCall(
+      kwarg.range(), *graph, aten::_set_item, {self, input_k, input_v}, {});
+    }
```

Comment:
This logic is very similar to 3076-3083, you might be able to factor it out to a helper lambda or function